src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Shield.shared.cs

```diff
@@ -240,10 +308,18 @@ protected override void OnPropertyChanged([CallerMemberName] string propertyName
 
 		void UpdateStatus() => ShieldStatus.Text = Status;
 
+		void UpdateSubjectColor() => ShieldSubjectContainer.BackgroundColor = SubjectBackgroundColor;
+
 		void UpdateColor() => ShieldStatusContainer.BackgroundColor = Color;
 
+		void UpdateStatusBackgroundColor() => ShieldStatusContainer.BackgroundColor = StatusBackgroundColor;
+
+		void UpdateSubjectTextColor() => ShieldSubject.TextColor = SubjectTextColor;
+
 		void UpdateTextColor() => ShieldStatus.TextColor = TextColor;
```

Comment:
We can already remove `UpdateTextColor` and let the old `TextColor` property update through `UpdateStatusTextColor`

Reply:
I preferred to leave it for now so that the later deletion won't be confusing.
How would `UpdateStatusTextColor()` knows which property has changed (StatusTextColor or TextColor) in order to assign it to ShieldStatus.TextColor ?

Reply:
> I preferred to leave it for now so that the later deletion won't be confusing.

I think the later deletion will be easier _and_ the code paths doing the same thing will be minimized. What happens when someone now updates the obsolete method but not the other one? We suddenly have a bug in code that we want to delete anyway.

> How would UpdateStatusTextColor() knows which property has changed (StatusTextColor or TextColor) in order to assign it to ShieldStatus.TextColor ?

This is not an issue, right? TextColor only updated the ShieldStatus, so we should just update that one and the behavior will be exactly the same as it is now.

Reply:
@jfversluis for 2nd point i still see it as an issue since we support both TextColor and StatusTextColor properties could be used by dev to change the same property, unless we change the method definition to take a parameter.

Reply:
In the [code that is implemented today](https://github.com/xamarin/XamarinCommunityToolkit/blob/ecd677a47678e3502aa4b892fd101e9f6b44bd11/src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Shield.shared.cs#L245), whenever you update `TextColor` it only updates `ShieldStatus.TextColor`. We don't want to break that behavior. To deduplicate code-paths we can remove `UpdateTextColor` and let that point to `UpdateStatusTextColor` that you are implementing here. The functionality will be exactly the same as it was before, but we don't need to worry about another codepath.

On the outside everything still looks the same, but on the inside it's easier to maintain for the time that we still support the old API